Add reference page for custom install (#3334)
knative/docs: docs/install/custom-install/custom-install.md

---
title: "Knative installation files and dependencies"
linkTitle: "Install files and dependencies"
weight: 20

> type: "docs"

Reviewer: If this used to be the files that was deleted please make sure you add in the aliases so that the redirects work

Reviewer: Also, I'd use "installation" consistently, rather than "install files"

Reviewer: I think we could probably actually just have a title here and no linkTitle and just call it "Installation files"; I don't think it actually includes any dependencies and not sure what is meant by this? Also I don't think there's a need to call out Knative 🙂

Author: The files were never published so do not need redirects. I was following the instructions in [#1869](https://github.com/knative/docs/issues/1896) when I gave the page a title but I agree that Installation files would be a more concise title.

---

This guide provides reference information about the YAML files in the Knative
Serving and Eventing releases.

The YAML files in the releases include:

- The custom resource definitions (CRDs) and core components required to install Knative.
- Optional components that you can apply to customize your installation.

> For how to install these files, see [YAML-based installation](../any-kubernetes-cluster.md).

Reviewer: ```suggestion
For information on installing these files, see [YAML-based installation](../any-kubernetes-cluster.md).
```

| serving-storage-version-migration.yaml | Migrates the storage version of Knative resources, including Service, Route, Revision, and Configuration, from `v1alpha1` and `v1beta1` to `v1`. Required by upgrade from version 0.18 to 0.19. | serving-core.yaml |

## Knative Eventing installation files

The table below describes the installation files available in the Knative Eventing release:

| File name | Description | Dependencies|
| --- | --- | --- |
| eventing-core.yaml | Knative Eventing core components. | eventing-crds.yaml |
| eventing-crds.yaml | Knative Eventing core CRDs. | none |
| eventing-post-install.yaml | Required jobs for upgrading to a new minor version. | eventing-core.yaml, eventing-crds.yaml |
| eventing-sugar-controller.yaml | Optional reconciler that watches for labels and annotations on certain resources to inject eventing components. | eventing-core.yaml |
| eventing.yaml | Combines `eventing-core.yaml`, `mt-channel-broker.yaml`, and `in-memory-channel.yaml`. | none |
| in-memory-channel.yaml | Components to configure In-Memory Channels. | eventing-core.yaml |

> | mt-channel-broker.yaml | Components to configure Multi-Tenant (MT) Channel Broker. | eventing-core.yaml |

Reviewer: I'd probably add a column to say whether these are optional since we mention above that some of it is extensions but we don't make it clear which files are optional in the table

Author: I decided to say in the description which ones are required because it is debatable which things are really optional.
